### PR TITLE
Chmod created hostpath directories to 777

### LINKED ISSUE #1990: Directories provisioned by hostPath provisioner are only writeable by root
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): bug report

Please provide the following details:

**Environment**:

**Minikube version** (use `minikube version`): v0.21.0
- **OS** (e.g. from /etc/os-release): ubuntu 17.04
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): virtualbox
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): minikube-v0.20.0.iso
- **Install tools**:
- **Others**:
 

**What happened**:

1. I provision a PVC
2. It is dynamically bound to a hostPath volume by the minikube provisioner
3. A pod is created that mounts the PVC
4. The process in the pod is running as uid 1000, with fsgid 1000 too
5. The process can not write to the PVC mount, since it is only writeable by root

Since we don't want to allow escalating privileges in the pod, we can't use the PVC mount at all.

**What you expected to happen**:

Some way of specifying in the PVC what uid / gid the hostPath should be owned by, so we can write to it.

**How to reproduce it** (as minimally and precisely as possible):

kubectl apply -f the following file:

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: test
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: "0"
---
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  volumes:
    - name: test 
      persistentVolumeClaim:
        claimName: test
  containers:
  - image: busybox:latest
    name: notebook
    volumeMounts: 
     - mountPath: /home/test
       name: test
    command: ["/bin/sh", "-c", "touch /home/test/hi"]
  securityContext:
    fsGroup: 1000
    runAsUser: 1000
```

It fails with the following output:
```
touch: /home/test/hi: Permission denied
```
If you set the fsGroup and runAsUser to 0, it succeeds.

### PR DESCRIPTION
**Pull request by yuvipanda**
Fixes #1990

### CODE REVIEW

**Comment by yuvipanda** [comment]
I'm not entirely sure if the defer for setting the umask back is good enough, or if I should try to set it back more immediately in the two obvious paths it needs to be set back

**Comment by r2d4** [comment]
@minikube-bot ok to test

**Comment by yuvipanda** [comment]
Can someone help me understand what the test failures mean? Am unsure...

**Comment by r2d4** [comment]
@yuvipanda sorry was related to me working on some of the integration servers.  

@minikube-bot retest this please

**Comment by yuvipanda** [comment]
Updated to use chmod per suggestion from @dlorenc

File: pkg/localkube/storage_provisioner.go [hunk 1]
```diff
@@ -73,6 +73,11 @@ func (p *hostPathProvisioner) Provision(options controller.VolumeOptions) (*v1.P
 		return nil, err
 	}
 
+	// Explictly chmod created dir, so we know mode is set to 0777 regardless of umask
+	if err := os.Chmod(path, 0777); err != nil {
```
Comment (dlorenc) [inline pkg/localkube/storage_provisioner.go]: Nice!

**Comment by yuvipanda** [comment]
Are these failures real?

**Comment by dlorenc** [comment]
Should be ok.

**Comment by yuvipanda** [comment]
Thank you very much!!!

**Comment by lawrencedudley** [comment]
Can I be a massive pain and ask if we can get a release with this in? It's causing me all sorts of headaches at the moment.

Nightly is fine but I can't find a way that's current where I can access the build for this PR.

**Comment by aaron-prindle** [comment]
We publish CI builds for each PR:
https://github.com/kubernetes/minikube/blob/master/docs/contributors/ci_builds.md

To access this build (for Mac) the link would be:
`https://storage.googleapis.com/minikube-builds/1991/minikube-darwin-amd64`

**Comment by lawrencedudley** [comment]
Thanks! I did see that - what threw me was I couldn't access 2025 for some reason so I assumed based on the directory index that these had stopped being published in February.

I've downloaded that - it seems to fix directory permissions but I'm still seeing weird issues with files that have been written into said directories. Am I mad or is that a possibility? Seems to only be after they've been written by a user in the container who is not root, i.e. write a file as a non-root user, then try to write to it again and you get a permissions issue.

I can replicate this with php -a and then file_put_contents to a file (first write is fine, subsequent one isn't).

Using vim, I get read-only file but then with the option to overwrite with :w!

Super confusing anyway :(

**Pull request merged**